pkg/api/controller.go

```diff
@@ -554,6 +556,24 @@ func (c *Controller) Login(w http.ResponseWriter, r *http.Request, body apigen.L
 	writeResponse(w, r, http.StatusOK, response)
 }
 
+func (c *Controller) STSLogin(w http.ResponseWriter, r *http.Request, body apigen.STSLoginJSONRequestBody) {
+	ctx := r.Context()
+	responseData, err := c.Authentication.ValidateSTS(ctx, body.Code, body.RedirectUri, body.State)
```

Comment:
I think we need to check here if external auth is enabled like we do with other calls

```
	if c.isExternalPrincipalNotSupported(ctx) {
		writeError(w, r, http.StatusNotImplemented, "Not implemented")
		return
	}
```